### PR TITLE
Add plugins' parsers to the `parser` option

### LINKED ISSUE #8389: Plugins don't add choices to the `parser` option
Plugins typically add parsers to be specified via the `parser` option, however there is no way for them to have their parsers listed as valid values of this option (in `prettier.getSupportedInfo().options`), which makes it difficult for editor integrations and other code to discover available parsers. It's still possible to find them in `prettier.getSupportedInfo().languages`, but it would make sense if Prettier did that itself and added choices to the option automatically.

**Steps to reproduce:**

1. `yarn add prettier @prettier/plugin-php`
2. `./node_modules/.bin/prettier --help` or `./node_modules/.bin/prettier --support-info`

**Expected behavior:**

The `parser` option should list `php` as a valid value.

**Actual behavior:**

It doesn't.

### PR DESCRIPTION
**Pull request by thorn0**
Fixes #8389

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests_integration/__tests__/__snapshots__/plugin-options-string.js.snap [hunk 1]
```diff
@@ -17,7 +17,16 @@ exports[` 1`] = `
                              How to handle whitespaces in HTML.
                              Defaults to css.
     --jsx-bracket-same-line  Put > on the last line instead of at a new line.
-                             Defaults to false."
+                             Defaults to false.
+    --jsx-single-quote       Use single quotes in JSX.
+                             Defaults to false.
+-   --parser <flow|babel|babel-flow|babel-ts|typescript|css|less|scss|json|json5|json-stringify|graphql|markdown|mdx|vue|yaml|html|angular|lwc>
++   --parser <flow|babel|babel-flow|babel-ts|typescript|css|less|scss|json|json5|json-stringify|graphql|markdown|mdx|vue|yaml|html|angular|lwc|foo-parser>
```
Comment (fisker) [inline tests_integration/__tests__/__snapshots__/plugin-options-string.js.snap:24]: I think we should display where this parser option from.

`--foo-string` need do that too.

Maybe seperate issue.

Comment (thorn0) [reply tests_integration/__tests__/__snapshots__/plugin-options-string.js.snap:24]: We can add the name of the originating plugin to the description of the value. Now the language name is use for it:

https://github.com/prettier/prettier/blob/93bdc0c61e19223c8818415756711aea38c15310/src/main/support.js#L115 

Should we change it from e.g. "PHP" to "PHP (@prettier/plugin-php plugin)"?

Comment (fisker) [reply tests_integration/__tests__/__snapshots__/plugin-options-string.js.snap:24]: Sounds good to me.

**Comment by fisker** [comment]
I tried install `@prettier/plugin-php` and `@prettier/plugin-pug`, so this info only available in `prettier --plugin @prettier/plugin-php --help parser`?


```console
node bin/prettier --plugin @prettier/plugin-php --help parser
--parser <flow|babel|babel-flow|babel-ts|typescript|css|less|scss|json|json5|json-stringify|graphql|markdown|mdx|vue|yaml|html|angular|lwc|php|pug>

  Which parser to use.

Valid options:

  flow             Flow
  babel            JavaScript
  babel-flow       Flow
  babel-ts         TypeScript
  typescript       TypeScript
  css              CSS
  less             Less
  scss             SCSS
  json             JSON
  json5            JSON5
  json-stringify   JSON.stringify
  graphql          GraphQL
  markdown         Markdown
  mdx              MDX
  vue              Vue
  yaml             YAML
  html             HTML
  angular          Angular
  lwc              Lightning Web Components
  php              PHP (plugin: @prettier/plugin-php)
  pug              Pug (plugin: @prettier/plugin-pug)
```

This looks good. I'll open another issue discuss about plugin options print in `prettier ---help` command.

**Pull request merged**